src/ObjCRuntime/Protocol.cs

```diff
@@ -24,6 +24,23 @@ public Protocol (string name)
 				throw new ArgumentException (String.Format ("'{0}' is an unknown protocol", name));
 		}
 
+		public Protocol (Type type)
+		{
+			var exception = new ArgumentException (string.Format ("'{0}' is an unknown protocol", type.FullName));
```

Comment:
This creates an exception instance for every protocol created (thru this .ctor). This is **not** efficient, use something like:

```
if (type.IsInterface) {
   foreach (var pa in type.GetCustomAttributes<ProtocolAttribute> (false)) {
      handle = objc_getProtocol (pa.Name);
      if (handle != IntPtr.Zero)
         return;
   }
}
if (handle == IntPtr.Zero)
   new ArgumentException (string.Format ("'{0}' is an unknown protocol", type.FullName));
```